lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,323 @@
+            # Calculate the total amount of memory required for all the VMs.
+            assert isinstance(node_requirements.memory_mb, int)
+            total_required_memory_mib += node_requirements.memory_mb
+
+        # Ensure host has enough memory for all the VMs.
+        if total_required_memory_mib > host_capabilities.free_memory_mib:
+            log.error(
+                f"Nodes require a total of {total_required_memory_mib} MiB memory. "
+                f"Host only has {host_capabilities.free_memory_mib} MiB free."
+            )
+            return False
+
+        return True
+
+    def _create_node_capabilities(self) -> schema.NodeSpace:
+        host_capabilities = self._host_capabilities
+        node_capabilities = schema.NodeSpace()
+        node_capabilities.name = "Hyper-V"
+        node_capabilities.node_count = 1
+        node_capabilities.core_count = search_space.IntRange(
+            min=1, max=host_capabilities.core_count
+        )
+        node_capabilities.disk = schema.DiskOptionSettings(
+            data_disk_count=search_space.IntRange(min=0),
+            data_disk_size=search_space.IntRange(min=1),
+        )
+        node_capabilities.network_interface = schema.NetworkInterfaceOptionSettings()
+        node_capabilities.network_interface.max_nic_count = 1
+        node_capabilities.network_interface.nic_count = 1
+        node_capabilities.gpu_count = 0
+        node_capabilities.features = search_space.SetSpace[schema.FeatureSettings](
+            is_allow_set=True,
+            items=[
+                schema.FeatureSettings.create(SerialConsole.name()),
+            ],
+        )
+
+        return node_capabilities
+
+    def _deploy_environment(self, environment: Environment, log: Logger) -> None:
+        self._deploy_nodes(environment, log)
+
+    def _prepare_common_vhd(self, vhd_local_path: PurePath, log: Logger) -> None:
+        if self._common_vhd_path:
+            return
+
+        vhd_remote_path = PureWindowsPath(
+            self.server_node.working_path / f"common_vhd.{vhd_local_path.suffix}"
+        )
+
+        is_zipped = False
+        if vhd_local_path.suffix == ".zip":
+            is_zipped = True
+            vhd_remote_path = PureWindowsPath(
+                self.server_node.working_path / "zipped_vhd.zip"
+            )
+
+        log.debug("Copying VHD to server")
+        self.server_node.shell.copy(vhd_local_path, vhd_remote_path)
+        log.debug("Finished copying VHD to server")
+
+        if is_zipped:
+            vhd_remote_path = self._unzip_vhd(vhd_remote_path)
+
+        self._common_vhd_path = vhd_remote_path
+
+    def _deploy_nodes(self, environment: Environment, log: Logger) -> None:
+        if environment.runbook.nodes_requirement is None:
+            return  # nothing to deploy?
+
+        test_suffix = "".join(random.choice(string.ascii_uppercase) for _ in range(5))
+        vm_name_prefix = f"lisa-{test_suffix}"
```

Comment:
Refer to Azure, generate vm name from the test env name or specified from runbook.

Reply:
Done.